File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hmap = {}

        for i, n in enumerate(nums):
            if target - n in hmap:
                return [hmap[target - n], i]
            else:
                hmap[n] = i

        return -1

File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    List<List<Integer>> res = new ArrayList<>();

    public List<List<Integer>> combinationSum(int[] nums, int target) {
        backTracking(0, new ArrayList<>(), nums, target);
        return res;
    }

    public void backTracking(int index, List<Integer> path, int[] nums, int target ) {
        if(target == 0) {
            res.add(new ArrayList<>(path));
            return;
        }
        if(target < 0) return;

        for(int i = index; i < nums.length; i++) {
            path.add(nums[i]);
            backTracking(i, path, nums, target - nums[i]);
            path.remove(path.size() - 1);
        }
    }
}
